### PR TITLE
Add KubeCarrier addon for KKP

### PR DESCRIPTION
**Pull request by rastislavs**
Signed-off-by: Rastislav Szabo <rastislav@kubermatic.com>

**What this PR does / why we need it**:
This PR adds KubeCarrier qddon for Kubermatic Kubernetes Platform.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Documentation**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: config/addon/addon-config.yaml [hunk 1]
```diff
@@ -0,0 +1,167 @@
+# Copyright 2020 The Kubermatic Kubernetes Platform contributors.
```
Comment (jiachengxu) [inline config/addon/addon-config.yaml:1]:
```suggestion
# Copyright 2020 The KubeCarrier Authors.
```

Comment (jiachengxu) [inline config/addon/addon-config.yaml:1]: Do we also want to create an `AddonConfig` for cert-manger plugin?

Comment (rastislavs) [reply config/addon/addon-config.yaml:1]: yup, good idea

Comment (jiachengxu) [reply config/addon/addon-config.yaml:1]: We use this `KubeCarrier Authors` in this project: https://github.com/kubermatic/kubecarrier/blob/master/LICENSE#L189-L201

**Pull request merged**